plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/hms/AbstractMetastoreTableOperations.java

```diff
@@ -113,7 +117,28 @@ protected final void commitNewTable(TableMetadata metadata)
         Table table = builder.build();
 
         PrincipalPrivileges privileges = owner.map(MetastoreUtil::buildInitialPrivilegeSet).orElse(NO_PRIVILEGES);
-        metastore.createTable(table, privileges);
+        try {
+            metastore.createTable(table, privileges);
+            cleanupOnFailure = false;
+        }
+        catch (SchemaNotFoundException
+               | TableAlreadyExistsException
+               | UnsupportedOperationException e) {
+            // Signal a non-retriable commit failure and eventually clean up metadata files corresponding to the current transaction
+            throw e;
+        }
+        catch (RuntimeException e) {
+            cleanupOnFailure = false;
+            // Let's retry to check if the commit has eventually got succeeded or not
```

Comment:
```suggestion
            // Let's retry to check if the commit has eventually got succeeded or not
```
     Check whether the creation of the table eventually succeeded
```

```